pkg/controller/atlasfederatedauth/atlasfederated_auth.go

```diff
@@ -19,20 +20,22 @@ func (r *AtlasFederatedAuthReconciler) ensureFederatedAuth(service *workflow.Con
 	}
 
 	// Get current IDP for the ORG
-	atlasFedSettings, _, err := service.SdkClient.FederatedAuthenticationApi.
+	atlasFedSettings, _, err := service.SdkClientSet.SdkClient20241113001.FederatedAuthenticationApi.
 		GetFederationSettings(service.Context, service.OrgID).
 		Execute()
 	if err != nil {
 		return workflow.Terminate(workflow.FederatedAuthNotAvailable, err.Error())
 	}
 
-	identityProvider, err := GetIdentityProviderForFederatedSettings(service.Context, service.SdkClient, atlasFedSettings)
+	identityProvider, _, err := service.SdkClientSet.SdkClient20241113001.FederatedAuthenticationApi.
+		GetIdentityProvider(service.Context, atlasFedSettings.GetId(), atlasFedSettings.GetIdentityProviderId()).
```

Comment:
After looking into the failed test. I remember this seems to be wrong.
Identity Provider ID present in the FedSettings is the Okta ID that was previously used as unique identifier.
Since `v20231115008` we must use the Atlas internal ID instead and that's why the deleted function was added (we have to list IdPs and match the OktaID). Check the internal IdP ID is present in latest FedSettings struct or undo the deletion of this function.

Reply:
I dont think it is present - reverted to the old function.